### PR TITLE
feat: docker-compose with opensearch

### PR DESCRIPTION
**Pull request by ghost**
- core
- camunda platform

image
image
image
image

### CODE REVIEW

File: docker-compose-core-opensearch.yaml [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+      start_period: 30s
+    env_file: connector-secrets.txt
+    networks:
+      - camunda-platform
+    depends_on:
+      - zeebe
+      - operate
+
+  opensearch:
+    image: opensearchproject/opensearch:2.16.0
```
Comment (igpetrov) [inline docker-compose-core-opensearch.yaml:116]: Can we move OS version to `.env` file?

**Review by igpetrov — APPROVED** [review]
LGTM

**Comment by igpetrov** [comment]
Hi @yevhen-cap 

It seems like OpenSearch cluster never turns green and always stays yellow. Can you please take a look?

File: docker-compose.opensearch.yaml [hunk 2]
```diff
@@ -0,0 +1,400 @@
...
+# https://docs.camunda.io/docs/self-managed/platform-deployment/kubernetes-helm/
+# For local development, we recommend using KIND instead of `docker-compose`:
+# https://docs.camunda.io/docs/self-managed/platform-deployment/helm-kubernetes/guides/local-kubernetes-cluster/
+
+# This is a full configuration with Zeebe, Operate, Tasklist, Optimize, Identity, Keycloak, and OpenSearch
+# See docker-compose-core.yml for a lightweight configuration that does not include Optimize, Identity, and Keycloak.
+
+services:
+
+  # opensearch-init:
```
Comment (igpetrov) [inline docker-compose.opensearch.yaml:14]: Can you please uncomment or remove?

File: docker-compose.opensearch.yaml
Comment (igpetrov) [inline docker-compose.opensearch.yaml:1]: I see that ES version also has Kibana. Do we need to have Kibana too?

**Comment by igpetrov** [comment]
> Hi @yevhen-cap
> 
> It seems like OpenSearch cluster never turns green and always stays yellow. Can you please take a look?

Likely related to this https://github.com/opensearch-project/opensearch-build/issues/4285

**Review by aabouzaid — CHANGES_REQUESTED** [review]
Adding a new entire Docker Compose file will increase the maintenance effort and duplication.
We need to do that via env var files instead of doing it with a new file.

File: .env [hunk 3]
```diff
@@ -48,3 +49,7 @@ RESOURCE_AUTHORIZATIONS_ENABLED=false
 MULTI_TENANCY_ENABLED=false
 
 OPENSEARCH_INITIAL_ADMIN_PASSWORD=OpenSearch_Admin1234!
+
+# search engine app: opensearch or elasticsearch
+XSEARCH=elasticsearch
+# XSEARCH=opensearch
```
Comment (igpetrov) [inline .env:55]: Maybe consider giving more vocal name what the variable is doing, e.g.
```suggestion
SEARCH_DB=elasticsearch
# SEARCH_DB=opensearch
```

File: docker-compose-core.yaml [hunk 4]
```diff
@@ -116,9 +124,48 @@ services:
...
+        hard: -1
+      nofile:
+        soft: 65536
+        hard: 65536
+    ports:
+      - "9200:9200"
+      - "9601:9600"
+    restart: always
+    healthcheck:
+      test: [ "CMD-SHELL", "curl -k -f http://localhost:9200/_cluster/health -u admin:${OPENSEARCH_INITIAL_ADMIN_PASSWORD} | grep -q -E 'yellow|green'" ]
```
Comment (igpetrov) [inline docker-compose-core.yaml:156]: Suggesting to leave a note why yellow is okay-ish
```suggestion
      # Single node OpenSearch clusters are considered 'yellow', see https://github.com/opensearch-project/opensearch-build/issues/4285
      test: [ "CMD-SHELL", "curl -k -f http://localhost:9200/_cluster/health -u admin:${OPENSEARCH_INITIAL_ADMIN_PASSWORD} | grep -q -E 'yellow|green'" ]
```

**Review by igpetrov — APPROVED** [review]
LGTM! Few optional comments.
Let's wait review/approval from @aabouzaid before we merge. Thank you!

File: docker-compose-core.yaml (hunk 4 above)
Comment (yevhen-cap) [reply docker-compose-core.yaml:156]: Agree with that suggestion

**Review by aabouzaid — APPROVED** [review]
LGTM, thank you :+1:

**Pull request merged**